### PR TITLE
fix: build warnings

### PR DESCRIPTION
**Pull request by Helias**
closes https://github.com/azerothcore/mod-costumes/issues/7

### CODE REVIEW

File: src/Costumes.h [hunk 1]
```diff
@@ -39,6 +39,7 @@ class Costumes : public PlayerScript, WorldScript
 
     // WorldScript
     void OnUpdate(uint32 diff) override;
+    void OnUpdate(Player *player, uint32 p_time) override;
```
Comment (r-o-b-o-t-o) [inline src/Costumes.h:42]: Could you move that a few lines up? They're sorted by script type, so it should be next to the other `PlayerScript` functions

File: src/Costumes.cpp [hunk 2]
```diff
@@ -54,7 +54,7 @@ struct PlayerState
     {
         if (morph)
         {
-            delete morph;
+            // delete morph;
```
Comment (r-o-b-o-t-o) [inline src/Costumes.cpp:57]: Let's fix this before merging 😛

**Review by r-o-b-o-t-o — CHANGES_REQUESTED** [review]

File: src/Costumes.h (hunk 1 above)
Comment (Helias) [inline src/Costumes.h:42]:
```suggestion
    void OnUpdate(Player *player, uint32 p_time) override;
    void OnUpdate(uint32 diff) override;
```

File: src/Costumes.h [hunk 3]
```diff
@@ -38,6 +38,7 @@ class Costumes : public PlayerScript, WorldScript
     void OnMapChanged(Player *player) override;
 
     // WorldScript
```
Comment (Helias) [inline src/Costumes.h:40]:
```suggestion
    // PlayerScript
```

File: src/Costumes.h [hunk 4]
```diff
@@ -37,7 +37,8 @@ class Costumes : public PlayerScript, WorldScript
     void OnPlayerEnterCombat(Player *player, Unit *enemy) override;
     void OnMapChanged(Player *player) override;
 
-    // WorldScript
+    // PlayerScript
+    void OnUpdate(Player *player, uint32 p_time) override;
     void OnUpdate(uint32 diff) override;
```
Comment (Helias) [inline src/Costumes.h:42]:
```suggestion
    // WorldScript
    void OnUpdate(uint32 diff) override;
```

File: src/Costumes.h [hunk 5]
```diff
@@ -37,6 +37,8 @@ class Costumes : public PlayerScript, WorldScript
     void OnPlayerEnterCombat(Player *player, Unit *enemy) override;
     void OnMapChanged(Player *player) override;
 
+    // PlayerScript
+    void OnUpdate(Player *player, uint32 p_time) override;
```
Comment (r-o-b-o-t-o) [inline src/Costumes.h:41]:
```suggestion
    void OnMapChanged(Player *player) override;
    void OnUpdate(Player *player, uint32 p_time) override;
```

**Review by r-o-b-o-t-o — CHANGES_REQUESTED** [review]

File: src/Costumes.h [hunk 6]
```diff
@@ -36,7 +36,7 @@ class Costumes : public PlayerScript, WorldScript
     bool CanUseItem(Player *player, ItemTemplate const *item, InventoryResult &result) override;
     void OnPlayerEnterCombat(Player *player, Unit *enemy) override;
     void OnMapChanged(Player *player) override;
-
+    void OnUpdate(Player *player, uint32 p_time) override;
     // WorldScript
```
Comment (r-o-b-o-t-o) [inline src/Costumes.h:40]:
```suggestion

    // WorldScript
```

File: src/Costumes.cpp (hunk 2 above)
Comment (Helias) [reply src/Costumes.cpp:57]: I didn't realize how to fix this :(

Comment (r-o-b-o-t-o) [reply src/Costumes.cpp]: @Helias can you see if the warning is gone now? I don't get this one with my Visual Studio I think

**Comment by Helias** [comment]
in the pipeline it does not appear, so it seems ok, but now the pipeline shows:
```
[ 21%] Building CXX object src/server/game/CMakeFiles/game.dir/AI/CreatureAIRegistry.cpp.o
/home/runner/work/mod-costumes/mod-costumes/modules/mod-costumes/src/Costumes.cpp:164:60: fatal error: format specifies type 'long long' but the argument has type 'int64' (aka 'long') [-Wformat]
            sprintf(formattedTime, "%02lld:%02lld:%02lld", hours, minutes, seconds);
                                    ~~~~~~                 ^~~~~
                                    %02ld
1 error generated.
make[2]: *** [modules/CMakeFiles/modules.dir/build.make:90: modules/CMakeFiles/modules.dir/mod-costumes/src/Costumes.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:1490: modules/CMakeFiles/modules.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
```

**Comment by Helias** [comment]
Locally with `lld` it's ok and with `ld` I get some warnings, in the pipeline is the opposite.

Probably the pipeline and my machine have different clang versions....

my clang version
```
Apple clang version 12.0.0 (clang-1200.0.32.29)
Target: x86_64-apple-darwin20.6.0
Thread model: posix
InstalledDir: /Library/Developer/CommandLineTools/usr/bin
```

**Review by r-o-b-o-t-o — APPROVED** [review]

**Comment by r-o-b-o-t-o** [comment]
Should we merge? Or do you want do dig deeper into the `lld` and `ld`?

**Comment by Helias** [comment]
let's merge it 💪

**Pull request merged**